### PR TITLE
USB for the nrf52 family

### PR DESCRIPTION
**Pull request by jamesadevine**

### CODE REVIEW

File: source/NRF52USB.cpp [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+    for (auto iface = cusb->interfaces; iface; iface = iface->next)
+    {
+        if (iface->out && iface->out->ep == ep)
+            return iface->out;
+    }
+    return NULL;
+}
+#endif
+
+#ifdef __cplusplus
```
Comment (mmoskal) [inline source/NRF52USB.cpp:81]: it's a .cpp file, you don't need this ifdef (of course still need the extern "C" though I would just put it in front of the IRQ handlers)

**Review by mmoskal — APPROVED** [review]

File: source/NRF52USB.cpp (hunk 1 above)
Comment (jamesadevine) [reply source/NRF52USB.cpp]: Thank you! this was a habit I picked up from working with the armcc compiler back when we were using it as part of micro:bit. (It compiles both C and C++ files in the same compile step)

File: source/NRF52USB.cpp [hunk 2]
```diff
@@ -0,0 +1,497 @@
+/*
+The MIT License (MIT)
+
+Copyright (c) 2017 Lancaster University.
```
Comment (finneyj) [inline source/NRF52USB.cpp:4]: update to 2021?

**Review by finneyj — APPROVED** [review]
Looks good - thanks James!

**Pull request merged**